apps/web/src/pages/settings/user-profile-page/useUserProfileSetPassword.ts

```diff
@@ -0,0 +1,58 @@
+import { showNotification } from '@mantine/notifications';
+import { IResponseError, PasswordResetFlowEnum } from '@novu/shared';
+import { api, useAuthContext } from '@novu/shared-web';
+import { useMutation } from '@tanstack/react-query';
+import { useCountdownTimer } from '../../../hooks';
+
+const COUNTDOWN_TIMER_START_SECONDS = 60;
+
+export const useUserProfileSetPassword = () => {
+  const { currentUser } = useAuthContext();
+  const {
+    timeRemaining: countdownTimerSeconds,
+    resetTimer,
+    stopTimer,
+  } = useCountdownTimer({
+    // start at 0 in the event the user opens the page directly to the set password flow via URL
+    startTimeSeconds: 0,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**thought (non-blocking):** This can go both ways. I see the reason why it has to be 0 by default. But this also allows the user to refresh the page to reset the timer every time.‏

Reply:
Ultimately we are handling a back-end concern solely through the front-end here, but we do have a safeguard for total number of password requests I believe. I agree it's not ideal, but we can see what comes up in UX feedback